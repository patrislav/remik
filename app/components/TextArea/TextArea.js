import React, {PropTypes} from 'react'

class TextArea extends React.Component {
  static propTypes = {
    onChange: PropTypes.func,
    value: PropTypes.string
  }

  static defaultProps = {
    onChange: () => {},
    value: ''
  }

  render() {
    /* eslint-disable no-unused-vars */
    const { value, onChange, ...props } = this.props
    return (
      <textarea onChange={::this._handleChange} value={value} {...props} />
    )
  }

  _handleChange(event) {
    this.props.onChange(event.target.value)
  }
}

export default TextArea
