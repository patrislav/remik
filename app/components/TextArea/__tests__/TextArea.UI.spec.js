import React from 'react'
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'

import TextArea from '../TextArea'

describe('<TextArea />', () => {
  it('renders correct UI with default props', () => {
    const wrapper = shallow(<TextArea />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('renders correct UI with provided value', () => {
    const wrapper = shallow(<TextArea value="VALUE" />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
