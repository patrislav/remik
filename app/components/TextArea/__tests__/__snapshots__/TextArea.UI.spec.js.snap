exports[`<TextArea /> renders correct UI with default props 1`] = `
<textarea
  onChange={[Function]}
  value="" />
`;

exports[`<TextArea /> renders correct UI with provided value 1`] = `
<textarea
  onChange={[Function]}
  value="VALUE" />
`;
